libs/gi/solver/src/GOSolver/BNBSplitWorker/polyUB.ts

```diff
@@ -329,6 +329,7 @@ function prodM(...monomials: Monomial[][]): Monomial[] {
   )
 }
 function foldLikeTerms(mon: Monomial[]): Monomial[] {
+  mon = mon.filter((m) => m.$k)
```

Comment:
Should line 344
```
if (mon[i].$k === 0) {
  mon.splice(i, 1)
  continue
}
```
be removed as well?

Reply:
I think its still needed to handle cases where `$k==0` appears after terms are combined, e.g. ax + -ax

Reply:
fair enough.